ledger/catchpointwriter.go

```diff
@@ -133,19 +137,19 @@ func (cw *catchpointWriter) WriteStep(ctx context.Context) (more bool, err error
 	}
 
 	// have we timed-out / canceled by that point ?
-	if more, err = hasContextDeadlineExceeded(ctx); more == true || err != nil {
+	if more, err = hasContextDeadlineExceeded(stepCtx); more == true || err != nil {
```

Comment:
The variable is called `more` to indicate that more calls to this function are needed in order to complete the writing process to the file, which was "interrupted" by the context deadline.

Reply:
Will it be fair to interpret this is maybe-more? 
Since hasContextDeadlineExceeded does not know that more calls to this function are needed. 
It might very well be that no more calls are needed, but it coincided with the timeout.

Reply:
If you'll look at the locations that call `hasContextDeadlineExceeded` you'll see that we basically always need more "time" in these cases. But we want to exit prematurely if the caller context exceed the timeout.